packages/manager/src/managers/slices/SlicesManager.ts

```diff
@@ -623,4 +651,55 @@ export class SlicesManager extends BaseManager {
 			variations,
 		};
 	}
+
+	private async _removeSliceFromCustomTypes(sliceID: string) {
+		const { models, errors: customTypeReadErrors } =
+			await this.customTypes.readAllCustomTypes();
+
+		models.forEach((customType) => {
+			const updatedJsonModel = Object.entries(customType.model.json).reduce(
+				(tabAccumulator, [tabKey, tab]) => {
+					const updatedTabFields = Object.entries(tab).reduce(
+						(fieldAccumulator, [fieldKey, field]) => {
+							if (
+								field.config === undefined ||
+								field.type !== "Slices" ||
+								field.config.choices === undefined
+							) {
+								return { ...fieldAccumulator, [fieldKey]: field };
+							}
+
+							const filteredChoices = Object.entries(
+								field.config.choices,
+							).reduce((choiceAccumulator, [choiceKey, choice]) => {
+								if (choiceKey === sliceID) {
+									return choiceAccumulator;
+								}
+
+								return { ...choiceAccumulator, [choiceKey]: choice };
+							}, {});
+
+							return {
+								...fieldAccumulator,
+								[fieldKey]: {
+									...field,
+									config: { ...field.config, choices: filteredChoices },
+								},
+							};
+						},
+						{},
+					);
+
+					return { ...tabAccumulator, [tabKey]: updatedTabFields };
+				},
+				{},
+			);
+
+			this.customTypes.updateCustomType({
+				model: { ...customType.model, json: updatedJsonModel },
+			});
```

Comment:
I think this needs to be awaited in case there's an error, you might want to use a `models.map(...)` + `Promise.all()` pattern like here: https://github.com/prismicio/slice-machine/blob/7e91e377ae4601e5584360b785b85273179d9c05/packages/manager/src/managers/SliceMachineManager.ts#L282-L283